nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RouteText.java

```diff
@@ -209,6 +215,30 @@
     private volatile Map<Relationship, PropertyValue> propertyMap = new HashMap<>();
     private volatile Pattern groupingRegex = null;
 
+    @VisibleForTesting
+    final static int PATTERNS_CACHE_MAXIMUM_ENTRIES = 10;
```

Comment:
We could probably cache more than 10 here. I think the idea on the PR was simply to convey that we need a reasonable upward bound, rather than allowing it to grow indefinitely. I would tend to lean more toward say 100 personally? Or even 1024 or so. A compiled Pattern is fairly small I believe in terms of heap utilization, so I wouldn't be concerned personally with such a limit.

Reply:
yes, I was thinking about introducing a configuration property for it but it seemed an overkill to me. I agree that it is not a big deal having more. I will set it to 1024, thanks.